agent/engine/execcmdagent/manager.go

```diff
@@ -0,0 +1,71 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+package execcmdagent
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	apicontainer "github.com/aws/amazon-ecs-agent/agent/api/container"
+	apitask "github.com/aws/amazon-ecs-agent/agent/api/task"
+	"github.com/aws/amazon-ecs-agent/agent/dockerclient/dockerapi"
+)
+
+const (
+	cmdField              = "CMD"
+	defaultRetryTimeout   = time.Minute * 10
+	defaultRetryMinDelay  = time.Second * 1
+	defaultRetryMaxDelay  = time.Second * 30
```

Comment:
ideally, we would want the retries to finish within the health checkup interval (which is 15 mins), so this might need to be changed accordingly.

Reply:
This timeout is the total timeout, not an individual timeout. I set it to 10 mins thinking about the 15 mins of the health check.

Reply:
my bad, I read it as 30 minutes 😅